Add support for single precision

as-amd opened the pull request:
This PR adds support to compile the RALFit in either double (default) or single precision.

Commit:
Finite Differences Jacobian

 * Adds support for FD in eval_J
 * Tweaks examples
 * Add copyright notice banners

Change-Id: I2ea721708a323c840f08af2d254b9c68e5f259a7

Commit:
Tidy output of Fortran example

Change-Id: Ia05dbe4d07d00f66b00fd7555497a3b3d415f34f

Commit:
Add type module

Add module to parametrize data types (int, real) in preparation
to build LP64/ILP64 and float/double variants

It also tweaks some of the examples

Change-Id: I8440c86be379839677bf65f5163e9d38cd17bb1f

Commit:
Minor updates to FD framework

Also fixes nlls_example2.c bug

Change-Id: Iec2ec4983eae41e229472d2bf902258a0a229d1d

Commit:
Fix type for integer inform counters

Change-Id: I45d8f565b07c4c7330317061db53ed09941f5e3e

Commit:
Add stand-alone entry-point to FD machinery

 * jacobian_setup,
 * jacobian_calc and
 * jacobian_free
 * example to show how to call this framework instead of using the
   solver's internal mecanism

Change-Id: I4eb24ce1cf67a78dceac541066f509657ef3d202

Commit:
Add stand-alone FD Jacobian example

Example to illustrate how to use RALFit FD
machinery to estimate a Jacobian matrix outside of
the solver

Change-Id: I34b81558e6375c8973f6988a65f7708354f3b391

Commit:
Fix bug is perturbation sign

Update and tidy examples

Change-Id: I094948f5705f836c7d2aa4905b6081ed6f0c6c06

Commit:
Add FD support for weighted residuals

Added new extended examples
Fixed bug in weight size passed in linesearch

Change-Id: Ia58f6c8bf74e7da3999b2c69e48fed874c95d112

Commit:
Add unit test

 * tweak examples
 * UT fd checker
 * UT fd solver

Change-Id: I8f8d4fb8674f7058a453400b1d66449f0ea08c4d

Commit:
Pretty print extremes for check derivative

Change-Id: I223cb5906f8853b7b346bef5a7ef18f2304a772e

Commit:
Reduce verbosity in derivative checker

Change-Id: Icdbf3b11671b17cc9bb2248a3e60ba07d2042630

Commit:
Remove ambiguous Public statement for derived type

Fixes ifort compilation error

Change-Id: Ib6ece887276e4a2700c79d26d224182edf0d3ec4

Commit:
Add Initial Fortran documentation

Still need to move common chunks to `../common`

Change-Id: Id611ab225488814649606de88bc3085c34f20689

tyronerees committed:
update the docs for python 3 and rtd (#120)

* update the docs for python 3 and rtd

* Fix warnings, including c issues

* fix typos and c signature

* Add a docs readme

* Fix python warnings/errors

Commit:
Add C documentation

Change-Id: Ibf48fa781dda37c4adc9cd581da0512dbc5177c3

Commit:
Minor fixes to C/Fortran documentation

 * Typos
 * Removal of trailing blanks

Change-Id: I85304c042b21728edfc544b12fa93b9a0d76eac7

Commit:
Add support for single precision

 * Add BLAS/LAPACK sources for proper mini-lapack compilation
 * Add Macros to compile source is single/double precisions
   (independently)
 * CMake dependency is still broken

Change-Id: I57ca466b76ef44285ddc78351bf4945f7c20c85e

Commit:
Remove hard-coded precision _d typo

Change-Id: I7f2c26b969b18835fbc93b25e51db684e737365c

Commit:
work-around to F90 CMake dependency-tree bug

Change-Id: I266bfe22df352e91999c019d73864cad59c79780

as-amd committed:
Merge pull request #2 from as-amd/refactor-cmake-deps

work-around to F90 CMake dependency-tree bug

Commit:
Make unique modules names

Make it possible to build in parallel once the library is built.

Change-Id: I722ef7737f9024082ab73d5614c5e5b62f334dc7

Commit:
Don't write to optional output var if not present

Change-Id: Icd0d27d82f5ea431bf28066980fbde4fd307d18e

Commit:
Cleaning up

    * Remove obsolete ComparisonTools folder
    * Remove unused functions
    * Restore maxit option for LanczosFD.f90
    * Don't force verbose make output on debug builds

Change-Id: Ic4fa828f400304487c1b7dbf94fa433eb8d7ad61

Commit:
Remove unused variable

Change-Id: I55a7da8342e2e6eae30b6a4fdd28b135a04037ea

Commit:
Refactor Abstract Interfaces

Move all fortran AB to a common place (nlls_types)

Change-Id: Ie416ba436ec37e9fcf0e632b797610c502322047

as-amd committed:
Merge branch 'master' into refactor

Commit:
Further cleaning

 * Remove c example source not compiled
 * fix f90 example typo

Change-Id: I65e415273fb1c0d600dd2a9fba4a12354e4b9058

Commit:
Refactor to allow in same library multi-precision objects

Change-Id: I9687416e926e011a9032a204528cfbe6ebe72827

Commit:
Bump up CMake version to handle FPP module names

Change-Id: I2f14e8de1a64c009f13a86e3d971b89af8591f2e

Commit:
Merge branch 'master' into refactor

Change-Id: Ib79119579cd5e4075ebf719a88da4a00f750ccc5

Commit:
Add missed file update (documentation)

Change-Id: Ifa6db4dbbf2f77db739fe50c8dd293fd0ccf169e

Commit:
Simplify examples

Examples avoid using #ifdef to work for both
precision types. Now they are simplified to use
one of the other and avoid sprinkles of #ifdef along
the examples.

It also takes care of updating the copyright banner

Change-Id: Ib741788a2615c904161c0e5003dee814f405b0b4

Commit:
Update READMEs

 * Main readme update broken links to logo and add link to
   secondary readme with details on compiling

 * secondary readme added more details on overall requirements,
   steps and a subsection on using accelerated compilers/libraries

Change-Id: I94cfbe5415b7fac385530c80f017d2be37e257e5

Commit:
Fix markdown typos

Change-Id: Ic651084da78f9ead53166da097062183147fe8a8

Commit:
Small fixes

 * Fix typos
 * move CMake flag "compile commands"
 * Add AMD links to AOCC and AOCL

Change-Id: Iad8980ebf4d7b1d7415dcf5dbc1cf114c50308e7

Commit:
Set default built type to "Release"

Also, in the README.md: Custom compiler section, replace the default flags
for release build.

Commit:
Clean CMakeLists.txt

Commit:
Merge branch 'master' into refactor

Commit:
Fix README typos

tyronerees committed:
Add the updated output to the C tests

Commit:
Add support for building with Ninja

tyronerees reviewed (changes requested):
Just a couple of minor things I've noticed. I've pushed a change to the generated C test output, which was causing the CI to fail.

Anvil doesn't seem to be kicking off jobs at the moment (our Cloud setup has been down due to networking issues, so I suspect that's why...) -- I'd just like to check that we can get the tests so that they pass next.

talassio committed:
Update libRALFit/README.md

Co-authored-by: Tyrone Rees <tyrone.rees@stfc.ac.uk>

talassio committed:
Update libRALFit/example/C/tracks.c

Co-authored-by: Tyrone Rees <tyrone.rees@stfc.ac.uk>

talassio committed:
Merge pull request #129 from ralna/master

Fix README.md UKRI logo (#127)

Commit:
Merge branch 'master' into refactor

Commit:
Update README.md

Commit:
Fix precision typo

Commit:
Fix Python cmake

Commit:
Internal fixes suppressions

Commit:
retrigger

Commit:
Fix typo

Commit:
Major updates to Cmake options

Commit:
Fix LIB string

Commit:
Update LIBS variable

Commit:
Add flags to known valid compilers

Commit:
Update compiler flags

Commit:
Fix NAG flags

Commit:
More flag fixes

Commit:
Fix Nagfor flag and parallel build

Commit:
Add NAGFOR flags to examples and ut

Commit:
further supressions for mini LAPACK

Commit:
More updates to mini LAPACK flags on Nagfor

Commit:
Add relative tol check as a CLI argument for UT

This change add some command line argument to allow as a fall-back
relaxation tolerance to check using relative tolerance and allow to pass
the test if it met. See also check_resvec subroutine.

Commit:
Pass CLI argumente to UT executable

Commit:
Fix CLI param typo

Commit:
Check for core dumps during testing

Commit:
Return flag on nagfor build

Commit:
Tidy

Commit:
Enable single precision build

Commit:
Retrigger job - test matrix filter

Commit:
Relax tolerance for ID:4 FD Solve in Single precision

Needs investigation on impoact non-ieee elemental function precision.
Perhaps for UT a simpler objective should be used? i.e., non-exponential model

Commit:
Revert relaxation on tolerance test

Commit:
Replace norm2 / dot intrinsics

gfortran and flang have slightly different
implementations and these produced some undesired diffs
in the UT. Replaced with calls to BLAS.

Note: Not all dot_product are replaced

Note: Matmul intrinsic should also be considered to be replaced.

This is a CRITICAL patch.

Commit:
Relax abs tol for resvec check

resvec norm2 C vs Fortran relax from 1e-8 to 2e-8
UT relative diff is still O(e-14) for norm2

Commit:
Tidy CMake config printing

Commit:
retrigger CI

Commit:
retrigger CI 2

tyronerees commented on libRALFit/pyproject.toml.in:
```diff
@@ -4,7 +4,7 @@ build-backend = 'setuptools.build_meta'
 name = 'RalNLLS'
-description = 'Nonlinear least squares'
-version = '1.0'
+description = 'RALFit a nonlinear least square solver'
```
```suggestion
description = 'RALFit a nonlinear least squares solver'
```

Commit:
Fix type square(s)

Commit:
Merge branch 'master' into refactor

Commit:
Add/update AMD copyright year
